src/ReactiveUI.Tests/Platforms/wpf/WpfActivationForViewFetcherTest.cs

```diff
@@ -40,6 +40,58 @@ public void FrameworkElementIsActivatedAndDeactivated()
             new[] { true, false }.AssertAreEqual(activated);
         }
 
+        [Fact]
+        public void WindowIsActivatedAndDeactivated()
+        {
+            var window = new WpfTestWindow();
+            var activation = new ActivationForViewFetcher();
+
+            var obs = activation.GetActivationForView(window);
+            obs.ToObservableChangeSet(scheduler: ImmediateScheduler.Instance).Bind(out var activated).Subscribe();
+
+            var loaded = new RoutedEventArgs();
+            loaded.RoutedEvent = FrameworkElement.LoadedEvent;
+
+            window.RaiseEvent(loaded);
+
+            new[] { true }.AssertAreEqual(activated);
+
+            window.Close();
+
+            new[] { true, false }.AssertAreEqual(activated);
+        }
+
+        [Fact]
+        public void WindowAndFrameworkElementAreActivatedAndDeactivated()
```

Comment:
`[Fact]
        public void WindowAndFrameworkElementAreActivatedAndDeactivated()
        {
            // The dispatcher thread
            var t = new Thread(() =>
            {
                var window = new WpfTestWindow();
                var uc = new WpfTestUserControl();

                window.RootGrid.Children.Add(uc);

                var activation = new ActivationForViewFetcher();

                var windowObs = activation.GetActivationForView(window);
                windowObs.ToObservableChangeSet(scheduler: ImmediateScheduler.Instance).Bind(out var windowActivated).Subscribe();

                var ucObs = activation.GetActivationForView(uc);
                ucObs.ToObservableChangeSet(scheduler: ImmediateScheduler.Instance).Bind(out var controlActivated).Subscribe();

                var loaded = new RoutedEventArgs();
                loaded.RoutedEvent = FrameworkElement.LoadedEvent;

                window.RaiseEvent(loaded);
                uc.RaiseEvent(loaded);

                new[] { true }.AssertAreEqual(windowActivated);
                new[] { true }.AssertAreEqual(controlActivated);

                window.Dispatcher.InvokeShutdown();

                new[] { true, false }.AssertAreEqual(windowActivated);
                new[] { true, false }.AssertAreEqual(controlActivated);
            });

            // Configure the thread
            t.SetApartmentState(ApartmentState.STA);
            t.Start();
            t.Join();
        }`

Please update the test to this code to run the test on the correct thread.

Reply:
I know there is an `XUnit.StaFact` package https://github.com/AArnott/Xunit.StaFact
Probably we could use this one for testing WPF controls etc.

Reply:
Thanks, I have updated the requested change to [STAFact]